### PR TITLE
FINERACT-2109: Lib dependencies and versions between modules

### PR DESCRIPTION
**Pull request by alberto-art3ch**
## Description

Java Lib dependencies and versions between modules

[FINERACT-2109](https://issues.apache.org/jira/browse/FINERACT-2109)

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/dependencies.gradle [hunk 1]
```diff
@@ -119,7 +119,7 @@ dependencies {
         exclude group: 'commons-logging'
     }
 
-    implementation 'io.swagger.core.v3:swagger-jaxrs2-jakarta:2.2.19'
```
Comment (adamsaghy) [inline fineract-provider/dependencies.gradle:122]: Wouldnt it be better to move versions into the `org.apache.fineract.dependencies.gradle` file? I was wondering whether it would be better to have only 1 configuration file that contains the versions.

File: fineract-doc/build.gradle [hunk 2]
```diff
@@ -40,7 +40,7 @@ asciidoctorj {
         pdf.version '1.6.2'
         diagram.version '2.2.1'
         epub.version '1.5.1'
-        revealjs.version '4.1.0'
```
Comment (adamsaghy) [inline fineract-doc/build.gradle:43]: By removing this dependency the ascii doc generation is still working?

File: fineract-provider/dependencies.gradle [hunk 3]
```diff
@@ -208,9 +208,9 @@ dependencies {
     testImplementation( 'io.cucumber:cucumber-spring',
             'io.github.classgraph:classgraph',
             'io.cucumber:cucumber-core',
-            'io.cucumber:cucumber-java',
-            'io.cucumber:cucumber-java8',
-            'io.cucumber:cucumber-junit-platform-engine',
+            'io.cucumber:cucumber-java:7.16.1',
```
Comment (adamsaghy) [inline fineract-provider/dependencies.gradle]: I think the versions should only be set in the `org.apache.fineract.dependencies.gradle`, would it be possible to move all dependency versions to a common place?

File: .gitignore [hunk 4]
```diff
@@ -25,8 +25,6 @@ licenses
 **/.asciidoctor/images/*
 **/.asciidoctor/diagram/*
 **/images/diag-*.svg
-fineract-avro-schemas/out/
```
Comment (adamsaghy) [inline .gitignore:28]: Why did you remove these lines?

File: buildSrc/build.gradle [hunk 5]
```diff
@@ -58,5 +58,4 @@ dependencies {
     implementation 'com.squareup.retrofit2:retrofit'
     implementation 'com.squareup.retrofit2:converter-jackson'
     implementation 'com.fasterxml.jackson.core:jackson-databind'
-    implementation 'com.squareup.okhttp3:okhttp'
```
Comment (adamsaghy) [inline buildSrc/build.gradle]: we dont need this anymore?

File: buildSrc/src/main/groovy/org.apache.fineract.dependencies.gradle [hunk 6]
```diff
@@ -76,13 +76,10 @@ dependencyManagement {
         dependency ('org.apache.tika:tika-parser-microsoft-module:2.9.2') {
             exclude 'org.bouncycastle:bcprov-jdk15on'
             exclude 'org.bouncycastle:bcmail-jdk15on'
-            exclude 'org.bouncycastle:bcprov-jdk18on'
```
Comment (adamsaghy) [inline buildSrc/src/main/groovy/org.apache.fineract.dependencies.gradle]: Do we need these dependencies for anything?

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Kindly see my review!

File: fineract-doc/build.gradle (hunk 2 above)
Comment (alberto-art3ch) [reply fineract-doc/build.gradle:43]: Yes, This is causing the missed library `okhttp-digest` that is not more available. Even the new version (4.0.0) for asciidoctor are still using the same dependency.

File: .gitignore (hunk 4 above)
Comment (alberto-art3ch) [reply .gitignore:28]: I removed all those line to set a more general line for the other modules.  

`**/out`

File: fineract-provider/dependencies.gradle (hunk 1 above)
Comment (alberto-art3ch) [reply fineract-provider/dependencies.gradle:122]: Done, readded

File: buildSrc/build.gradle (hunk 5 above)
Comment (alberto-art3ch) [reply buildSrc/build.gradle]: Done! readded

File: .gitignore (hunk 4 above)
Comment (adamsaghy) [reply .gitignore:28]: `fineract-provider/src/main/generated/` is not matching for `**/out`. Please fix it

Comment (adamsaghy) [reply .gitignore:28]: Also what is `gradleExp`? Please remove

File: fineract-doc/build.gradle (hunk 2 above)
Comment (adamsaghy) [reply fineract-doc/build.gradle:43]: My question was whether ascii doc generation is still working even if this dependency got removed?

Comment (alberto-art3ch) [reply fineract-doc/build.gradle:43]: Yes, the finerac-doc build process is stil working

File: fineract-provider/dependencies.gradle (hunk 3 above)
Comment (alberto-art3ch) [reply fineract-provider/dependencies.gradle]: Done! updated

**Review by alberto-art3ch — COMMENTED** [review]
Done! PR updated

**Review by adamsaghy — APPROVED** [review]
LGTM
